src/js/components/DataTable/stories/MultiplePin.js

```diff
@@ -0,0 +1,331 @@
+    pinned: 7529511627776,
+    savings: [
+      { unit: 'TiB', value: 8.0 },
+      { unit: 'xGHz', value: 333.2 },
+    ],
+  },
+];
+
+const columns = [
+  {
+    property: 'arrays',
+    header: 'Arrays',
+    render: datum => <Text truncate>{datum.arrays}</Text>,
+    sortable: false,
+  },
+  {
+    property: 'size',
+    header: (
+      <Text color="text-strong" weight="bold">
+        Size{' '}
+        <Text size="xsmall" weight="normal" color="text">
+          (TiB)
+        </Text>
+      </Text>
+    ),
+    render: datum =>
+      // bytes to tebibytes
+      (datum.size / 2 ** 40).toFixed([1]),
+    align: 'end',
+  },
+  {
+    property: 'pinnable',
+    header: (
+      <Text color="text-strong" weight="bold">
+        Pinnable{' '}
+        <Text size="xsmall" weight="normal" color="text">
+          (B)
+        </Text>
+      </Text>
+    ),
+    render: datum =>
+      // bytes to tebibytes
+      (datum.pinnable / 2 ** 40).toFixed([1]),
+    align: 'end',
+  },
+  {
+    property: 'pinned',
+    header: (
+      <Box pad={{ horizontal: 'small', vertical: 'xsmall' }}>
+        <Text color="text-strong" weight="bold">
+          Pinned{' '}
+          <Text size="xsmall" weight="normal" color="text">
+            %
+          </Text>
+        </Text>
+      </Box>
+    ),
+    render: datum => (
+      <Box pad={{ vertical: 'xsmall' }}>
+        <Meter
+          values={[{ value: datum.pinned / datum.pinnable, color: 'graph-2' }]}
+          max={1}
+          thickness="small"
+          size="small"
+        />
+      </Box>
+    ),
+    sortable: false,
+  },
+  {
+    property: 'savings',
+    header: (
+      <Text color="text-strong" weight="bold">
+        Savings{' '}
+        <Text size="xsmall" weight="normal" color="text">
+          (xGHz)
+        </Text>
+      </Text>
+    ),
+    align: 'end',
+    render: datum => (
+      <Text truncate>{datum.savings[1] && `${datum.savings[1].value}`}</Text>
+    ),
+  },
+];
+
+const handleClickRow = obj => {
+  // eslint-disable-next-line no-alert
+  alert(`
+  Record was clicked:
+  {
+      id: ${obj.id},
+      poolName: ${obj.poolName}
+  }
+
+  You can use onClickRow() to navigate to a record's detail
+  page, open a panel or modal to edit the record, or perform
+  other actions as you see fit.
+  `);
+};
+
+const MultiplePins = () => {
+  const size = React.useContext(ResponsiveContext);
+  return (
+    <>
+      <Box>ResponsiveContext size: {size}</Box>
+      <Box>
+        Columns `id`, `Pool Name` and `Group Name` will become pinned columns
+        once the current ResponsiveContext size is `small`
+      </Box>
+      <Box height="medium" overflow="auto">
+        <DataTable
+          data={data}
+          columns={[
+            {
+              property: 'id',
+              header: 'Id',
+              primary: true,
+              render: datum => datum.id.slice(datum.id.length - 5),
+              pin: size === 'small',
```

Comment:
```suggestion
              pin: true,
```
Wondering if it'd be better to set the pin to `true` and then add a fixed width to the containing Box (`width="large"`) so people consuming the example won't have to resize their window to see the functionality.

Reply:
Would need adjusting on the two columns below this also.